src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1572,7 +1572,7 @@ public bool Equals(WeakReference x, WeakReference y)
 
 			public int GetHashCode(WeakReference obj)
 			{
-				return obj?.GetHashCode() ?? 0;
+				return obj?.Target?.GetHashCode() ?? 0;
```

Comment:
Relevant to this PR? Should be a separate commit at least

Reply:
Right, It's not relevant, I removed it and will do another PR for that.